src/Elliptic/Systems/Poisson/Tags.hpp

```diff
@@ -24,18 +20,19 @@ namespace Poisson {
 /*!
  * \brief The scalar field \f$u(x)\f$ to solve for
  */
+template <typename T>
 struct Field : db::SimpleTag {
-  using type = Scalar<DataVector>;
+  using type = Scalar<T>;
```

Comment:
Some systems have explicit `DataVector` tags and others template the tags on the data type. What should we go with generally? The `double` specialization is really only useful for tests I suppose, so not sure if it's worth it.

Reply:
I would go with `DataVector` unless you have a reason not to.  The main potential use case for `double` so far has been for lensing or tracer particles, but I have yet to be fully convinced that we can't just use `DataVector`s for them as well since unlike SpEC, a `DataVector` doesn't require a `Mesh`.  Another potential use case for `double` is for functions that need to be called within an iterative algorithm (such as root finding) where it only makes sense to do something point by point.  I would think only `DataVector` would be fine for the Poisson system

Reply:
So for the `NewtonianEuler` system we template on the data type for this reason (tracer particles)?